dali/pipeline/operators/geometric/bb_flip.h

```diff
@@ -15,12 +15,43 @@
 #ifndef DALI_PIPELINE_OPERATORS_GEOMETRIC_BB_FLIP_H_
 #define DALI_PIPELINE_OPERATORS_GEOMETRIC_BB_FLIP_H_
 
-#include <dali/pipeline/operators/operator.h>
 #include <dali/pipeline/operators/common.h>
+#include <dali/pipeline/operators/operator.h>
 #include <string>
 
 namespace dali {
 
+class BoundingBox {
+ public:
+  explicit BoundingBox(float x1, float y1, float w1, float h1, bool ltrb = false)
+      : x{x1}, y{y1}, w{ltrb ? w1 - x1 : w1}, h{ltrb ? h1 - y1 : h1} {
+    DALI_ENFORCE(x >= 0, "Expected x >= 0. Received: " + std::to_string(x));
```

Comment:
That's kind of censorship of user input. We may (and must) enforce these constraints when we slice/crop/... based on those bounding boxes - but as long as they are just data that will be fed into a network, why would we interfere? I'd only check that width and height are non-negative.

Reply:
Frameworks such as Tensorflow appear to put constraints on bboxes being between [0.0, 1.0]. See below:

https://www.tensorflow.org/api_docs/python/tf/image/sample_distorted_bounding_box
https://www.tensorflow.org/api_docs/python/tf/image/draw_bounding_boxes

While I acknowledge a number of cases where this constraint might be a limiting factor (detection of objects outside of the FOV), i think the most reasonable thing to do at the moment is to stick to the scope of SSD. Once we have a solid baseline, we can decide how to enable these other use cases (several ideas come to mind). IMHO it will ultimately depend on how we want to interact with different frameworks, but by now I don't consider this censorship of user input, but rather consistency with existent frameworks.

Reply:
I must agree with Pablo, let us stick now with augmentations that other FWs do.